hdbt.theme

```diff
@@ -691,8 +691,30 @@ function hdbt_preprocess_paragraph(array &$variables) {
   // and tell its children the heading level.
   if ($paragraph_type == 'accordion_item') {
     $paragraph_parent = $paragraph->getParentEntity();
-    $accordion_item_heading_level = $paragraph_parent->get('field_accordion_heading_level')->getString();
-    $variables['accordion_item_heading_level'] = $accordion_item_heading_level;
+    $accordion_has_title = (bool) !$paragraph_parent
+      ->get('field_accordion_title')->isEmpty();
+    $accordion_title_level = $paragraph_parent
+      ->get('field_accordion_title_level')->getString();
+    $accordion_item_heading_level = $paragraph_parent
+      ->get('field_accordion_heading_level')->getString();
+
+    // Check if accordion paragraph has a title and set item heading level
+    // based on that. If not, use level given by editor.
+    if ($accordion_has_title) {
+      // Remove inaccessible skipping between title level and item level.
+      // For example:
+      //   title h3, item h6 --> fixed item h4
+      //   title h2, item h3 --> item h3
+      //   title h3, item h3 --> item h3
+      //   title h5, item h3 --> item h3
+      if ($accordion_title_level + 1 < $accordion_item_heading_level) {
+        $variables['accordion_item_heading_level'] = $accordion_title_level + 1;
+      } else {
+        $variables['accordion_item_heading_level'] = $accordion_item_heading_level;
+      }
+    } else {
```

Comment:
Functionality wise seems to be working as described.

There are few coding standard issues here that I tried to fix in the suggestion:
- extra indenting is not allowed (except maybe in a todo block) so I removed them
- single-line comments should end with some specific characters so I added commas and a full stop.
- `else` should start on a new line

```suggestion
      // title h3, item h6 --> fixed item h4,
      // title h2, item h3 --> item h3,
      // title h3, item h3 --> item h3,
      // title h5, item h3 --> item h3.
      if ($accordion_title_level + 1 < $accordion_item_heading_level) {
        $variables['accordion_item_heading_level'] = $accordion_title_level + 1;
      }
      else {
        $variables['accordion_item_heading_level'] = $accordion_item_heading_level;
      }
    }
    else {
```

Reply:
Thanks, sometimes I wonder the benefit of these linter comment settings. Fixed.